src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/CodeBlockService.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Razor;
+internal static class CodeBlockService
+{
+    /// <summary>
+    ///  Creates a <see cref="TextEdit"/> that will place the formatted generated method within a @code block in the file.
+    /// </summary>
+    /// <param name="code">The <see cref="RazorCodeDocument"/> of the file where the generated method will be placed.</param>
+    /// <param name="generatedMethod">
+    ///  The skeleton of the generated method where a '0' should be placed anywhere that needs to have some initial indenting, '1' for the inner indent of the method, and a '2' for where the method name should be placed.
+    ///  It should look something like: 0public void 2(){\r\n1throw new NotImplementedException();\r\n0}
+    /// </param>
+    /// <param name="methodName">The name of the method that is being generated.</param>
+    /// <returns>A <see cref="TextEdit"/> that will place the formatted generated method within a @code block in the file.</returns>
+    public static TextEdit CreateFormattedTextEdit(RazorCodeDocument code, string generatedMethod, string methodName)
+    {
+        var indentSize = code.GetCodeGenerationOptions().IndentSize;
+        var csharpCodeBlock = code.GetSyntaxTree().Root.DescendantNodes().Where(e => e.Kind == SyntaxKind.CSharpCodeBlock && e.GetContent().StartsWith("@code") && e.GetContent().Contains('{') && e.GetContent().Contains('}')).FirstOrDefault();
```

Comment:
Calling `GetContent` repeatedly is not a good idea, it will create a new string each time, and copy the contents from the tree etc. If we end up having to call GetContent at all, definitely only do it once and re-use the result. I think in this case though, we can be a little smarter about finding code blocks by using the nodes themselves. Check out what we do in RazorFormattingPass:
https://github.com/dotnet/razor/blob/main/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/RazorFormattingPass.cs#L113-L117

(ignore the fact that the code I linked to calls GetContent twice - it shouldn't!)

Reply:
Extra note, once you're using nodes, notice that code also does things like call `TryGetOpenBrace` etc. If those methods aren't in a shared location, would be ideal to share them, so they can be reused here, at the end of the method where you're doing a bunch of work to find the open and close brace positions etc. With the right helpers navigating the tree, they could be just a few `node.SpanStart` calls